test/integration_tests/executor_test.go

```diff
@@ -90,3 +90,15 @@ func TestUpdate(t *testing.T) {
 		assert.Fail(t, "err is not of type GenericSwaggerError")
 	}
 }
+
+func TestStartWorkflowWithContext(t *testing.T) {
+	executor := testdata.WorkflowExecutor
+
+	ctx, cancel := context.WithCancel(context.Background())
+	// cancel straightaway on purpose
+	cancel()
+
+	_, err := executor.StartWorkflowWithContext(ctx, &model.StartWorkflowRequest{})
```

Comment:
An example test that shows that `StartWorkflowWithContext` throws an error if called with a cancelled context.